### ISSUE TITLE
How to set learning rate decay on sklearn interface like `LGBMClassifier`

### ISSUE THREAD

**Issue by BovenPeng** [microsoft/LightGBM#2698]
## Summary

I have searched some relevant information before, but I only get the implement of learning rate decay in LightGBM original interface like follows:
https://github.com/Microsoft/LightGBM/issues/129
https://lightgbm.readthedocs.io/en/latest/Parameters.html
https://www.kaggle.com/c/talkingdata-adtracking-fraud-detection/discussion/56158
```
gbm = lgb.train(
    params,
    lgb_train,
    num_boost_round=10,
    init_model=gbm,
    valid_sets=lgb_eval,
    callbacks=[lgb.reset_parameter(learning_rate = lambda current_round: alpha * e**(k*current_round)
)
```
And the parameter `refit_decay_rate` controls the `leaf_output`, which is kind of like to avoid overfitting. 
Sorry that I didn't find some useful relevant information about it so I was not sure of its use.

## Motivation

So I want to know there is possible to implement learning rate decay in the sklearn interface, `LGBMClassifier()`, like the lgb original code example before.
Thanks in advance.

**Comment by StrikerRUS** [comment]
Hi @BovenPeng !

Sure, you can write learning rate decay for the sklearn wrapper just like for the original interface:
```
est = lgb.LGBMClassifier().fit(X, y, callbacks=[lgb.reset_parameter(learning_rate=lambda current_round: alpha * e**(k * current_round))])
```
https://github.com/microsoft/LightGBM/blob/dd16fa99fca93c4c47d75d1d41d331d5ca6f1c33/python-package/lightgbm/sklearn.py#L445-L447

``
